dbms/src/Storages/DeltaMerge/DeltaMergeStore_Ingest.cpp

```diff
@@ -747,10 +747,14 @@ UInt64 DeltaMergeStore::ingestFiles(
     Segments updated_segments;
     if (!range.none())
     {
-        if (use_split_replace)
-            updated_segments = ingestDTFilesUsingSplit(dm_context, range, external_files, files, clear_data_in_range);
-        else
-            updated_segments = ingestDTFilesUsingColumnFile(dm_context, range, files, clear_data_in_range);
+        if (!segments.empty() || !external_files.empty())
```

Comment:
I think here accessing `segments` without any lock protection will cause data race in corner cases @CalvinNeo

Reply:
Will fix in https://github.com/pingcap/tiflash/pull/8245